source/microbit/microbitimage.cpp

```diff
@@ -347,27 +271,45 @@ STATIC mp_obj_t microbit_image_make_new(const mp_obj_type_t *type_in, mp_uint_t
     }
 }
 
-STATIC microbit_image_obj_t *image_crop(microbit_image_obj_t *img, mp_int_t x, mp_int_t y, mp_int_t w, mp_int_t h) {
+STATIC void image_blit(microbit_image_obj_t *src, greyscale_t *dest, mp_int_t x, mp_int_t y, mp_int_t w, mp_int_t h, mp_int_t xdest, mp_int_t ydest) {
     if (w < 0)
         w = 0;
     if (h < 0)
         h = 0;
-    greyscale_t *result = greyscale_new(w, h);
-    mp_int_t intersect_x0 = max(0, x);   
-    mp_int_t intersect_y0 = max(0, y);
-    mp_int_t intersect_x1 = min(img->width(), x+w);
-    mp_int_t intersect_y1 = min(img->height(), y+h);
-    /* If the cropped image is larger than the intersection then
-     * make sure that all other pixels are set to 0 */
-    if (w > intersect_x1 - intersect_x0 || h > intersect_y1 - intersect_y0) {
-        result->clear();
+    mp_int_t intersect_x0 = max(0, xdest);
+    mp_int_t intersect_y0 = max(0, ydest);
+    mp_int_t intersect_x1 = min(dest->width, xdest+w);
+    mp_int_t intersect_y1 = min(dest->height, ydest+h);
+    for (int i = intersect_x0; i < intersect_x1; ++i) {
+        for (int j = intersect_y0; j < intersect_y1; ++j) {
+            dest->setPixelValue(i, j, 0);
+        }
```

Comment:
If I'd remembered to commit the test, that would have been more obvious :man_facepalming: